src/crystal/project/git_dependency.cr

```diff
@@ -0,0 +1,60 @@
+require "crystal/project/dependency"
+require "crystal/project/project_error"
+
+module Crystal
+  class GitDependency < Dependency
+    getter target_dir
+
+    def initialize(@repo, name = nil : String?, branch = nil : String?)
+      unless repo =~ /(.*)(:|\/)(.*)\/(.*)(.git)/
```

Comment:
Being smart here will fail, `author/repo` is largely a Github thing, not a git thing. IMO this class should just take the URL and branch and do the git commands, then normalize the name into something that's a valid directory name. Yes, that will invalidate the names of the current github dependencies, but we're still early, we can introduce breaking changes such as this.